bundles/statistics/statsgrid2016/view/SearchFlyout.js

```diff
@@ -85,12 +85,21 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.view.SearchFlyout', function (t
             }
 
             var newActiveIndicator = false;
-
+            
             selectedIndicators.forEach(function (indicator) {
+                var added;
                 if (indicator === '') {
                     return;
                 }
-                var added = me.service.getStateService().addIndicator(values.datasource, indicator, values.selections);
+                if (Array.isArray(values.selections.time)) {
```

Comment:
Would it be a better solution to just check if any of the selections have an array as value? Then you wouldn't need to care if it's time.

Reply:
We still need to know the time values when they are in an array because then we need to call addIndicator in a loop for each of those values.

Reply:
Can't we just loop when we encounter a value that's array? addIndicator() doesn't care if there's a time variable.